pkg/ec2metadata/ec2metadata.go

```diff
@@ -196,16 +196,17 @@ func (e *Service) GetRebalanceRecommendationEvent() (rebalanceRec *RebalanceReco
 // GetMetadataInfo generic function for retrieving ec2 metadata
 func (e *Service) GetMetadataInfo(path string) (info string, err error) {
 	resp, err := e.Request(path)
-	if resp != nil && (resp.StatusCode < 200 || resp.StatusCode >= 300) {
-		return "", fmt.Errorf("Metadata request received http status code: %d", resp.StatusCode)
-	}
 	if err != nil {
 		return "", fmt.Errorf("Unable to parse metadata response: %w", err)
 	}
 	defer resp.Body.Close()
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return "", fmt.Errorf("Unable to parse http response: %w", err)
+		return "", fmt.Errorf("Unable to parse http response. Status code: %d. %w", resp.StatusCode, err)
+	}
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
```

Comment:
Can we add back the `resp != nil` check before accessing `StatusCode`? I know `resp` can't be nil if we add the update from my other comment to `Request()`, but not 100% certain without that change..

Reply:
If `resp` is nil, we would get an error on line 202 or 203 before even getting here, so I don't think an additional nil check here would be very useful. I'll add it back to line 199 instead so we can catch it before trying to reference any properties.

Reply:
> If resp is nil, we would get an error on line 202 or 203 before even getting here

Can you help me understand where the error gets assigned when resp is nil? Looking thru the code it seems like it's possible to return nil response and no error. Unless the `Request()` method has been refactored recently, I think the previous `!= nil` check is still necessary.

> I'll add it back to line 199 instead so we can catch it before trying to reference any properties.

Idk if I'd say idiomatic, but usually common to see the `!= nil` check coupled with the use (as it was previously).

Reply:
Discussed offline. The new `resp != nil` check should address this concern.